api/envoy/extensions/transport_sockets/tls/v3/common.proto

```diff
@@ -253,7 +254,26 @@ message CertificateProviderPluginInstance {
   string certificate_name = 2;
 }
 
-// [#next-free-field: 15]
+// Matcher for subject alternative names, to match both type and value of the SAN.
+message SubjectAltNameMatcher {
+  // Indicates the choice of GeneralName as defined in section 4.2.1.5 of RFC 5280 to match
+  // against.
+  enum SanType {
+    SAN_TYPE_UNSPECIFIED = 0;
+    EMAIL = 1;
+    DNS = 2;
+    URI = 3;
+    IP_ADDRESS = 4;
+  }
+
+  // Specification of type of SAN. Note that the default enum value is an invalid choice.
+  SanType san_type = 1 [(validate.rules).enum = {in: 1 in: 2 in: 3 in: 4}];
```

Comment:
nit: `in: [1,4]`

Reply:
[1, 4] means only 1 and 4 and not 1 to 4. I had [1, 2, 3, 4] but that failed format in CI

Reply:
Maybe: `[(validate.rules).enum = {defined_only: true not_in: 0}]` would work here.